dotCMS/src/main/java/com/dotcms/rendering/velocity/services/FieldLoader.java

```diff
@@ -0,0 +1,99 @@
+
+import org.apache.velocity.runtime.resource.ResourceManager;
+
+/**
+ * @author Jason Tesser
+ * @since 1.6.5
+ *
+ */
+public class FieldLoader implements DotLoader {
+	public void invalidate(String fieldInode, String contentletIdent, boolean EDIT_MODE,String filePath) throws DotDataException, DotSecurityException {
+		removeFieldFile(fieldInode, contentletIdent, EDIT_MODE);
+	}
+	
+	public InputStream buildVelocity(String fieldInode, String contentInode, boolean EDIT_MODE,String filePath) throws DotDataException, DotSecurityException {
+		InputStream result;
+		Field field = APILocator.getContentTypeFieldAPI().find(fieldInode);
+		if(!UtilMethods.isSet(field)){
+			Logger.warn(this.getClass(),"Field not found.  Unable to load velocity code");
+			return new ByteArrayInputStream("".toString().getBytes());
+		}
+		ContentletAPI conAPI = APILocator.getContentletAPI();
+		Contentlet content = conAPI.find(contentInode, APILocator.getUserAPI().getSystemUser(), true);
+		if(!UtilMethods.isSet(content)){
+			Logger.warn(this.getClass(),"Content not found.  Unable to load velocity code");
+			return new ByteArrayInputStream("".toString().getBytes());
+		}
+		Object contFieldValueObject = conAPI.getFieldValue(content, field);
+		String contFieldValue = "";
+		
+		
+		
+		if(field instanceof ConstantField || field instanceof HiddenField){
+			contFieldValue = field.values() == null ? "" : field.values();
+		}else{
+			contFieldValue = contFieldValueObject == null ? "" : contFieldValueObject.toString();
+		}
+		
+		if(contFieldValue != null && contFieldValue.endsWith("#")){
+			contFieldValue = contFieldValue.substring(0, contFieldValue.length()-1);
+			contFieldValue += "$esc.h";
+		}
+		
+
+        return writeOutVelocity(filePath, contFieldValue);
+
+
+	}
+	
+	public void removeFieldFile (String fieldInode, String contentInode, boolean EDIT_MODE) {
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        velocityRootPath += java.io.File.separator;
+        String folderPath = (!EDIT_MODE) ? "live" + java.io.File.separator: "working" + java.io.File.separator;
+        String filePath=folderPath + contentInode + "_" + fieldInode + "." + VelocityType.FIELD.fileExtension;
+        java.io.File f  = new java.io.File(velocityRootPath + filePath);
+        f.delete();
+        DotResourceCache vc = CacheLocator.getVeloctyResourceCache();
+        vc.remove(ResourceManager.RESOURCE_TEMPLATE + filePath );
+    }
+
+    @Override
+    public InputStream writeObject(String id1, String id2, boolean live, String language, String filePath)
+            throws DotDataException, DotSecurityException {
+
+            return this.buildVelocity(id2, id1, !live, filePath);
+        
+    }
+
+    @Override
+    public void invalidate(Object obj) {
+        // TODO Auto-generated method stub
+        
+    }
+
+    @Override
+    public void invalidate(Object obj, boolean live) {
+        // TODO Auto-generated method stub
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)